addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -0,0 +1,277 @@
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.somfytahoma.handler.SomfyTahomaBaseThingHandler;
+import org.openhab.binding.somfytahoma.handler.SomfyTahomaBridgeHandler;
+import org.openhab.binding.somfytahoma.internal.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.openhab.binding.somfytahoma.SomfyTahomaBindingConstants.*;
+
+/**
+ * The {@link SomfyTahomaItemDiscoveryService} discovers rollershutters and
+ * action groups associated with your TahomaLink cloud account.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+public class SomfyTahomaItemDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyTahomaItemDiscoveryService.class);
+    private SomfyTahomaBridgeHandler bridge;
+    private DiscoveryServiceCallback discoveryServiceCallback;
+    private ScheduledFuture<?> discoveryJob;
+
+
+    private static final int DISCOVERY_TIMEOUT_SEC = 10;
+    private static final int DISCOVERY_REFRESH_SEC = 30;
+
+    public SomfyTahomaItemDiscoveryService(SomfyTahomaBridgeHandler bridgeHandler) {
+        super(DISCOVERY_TIMEOUT_SEC);
+        logger.debug("Creating discovery service");
+        this.bridge = bridgeHandler;
+    }
+
+    /**
+     * Called on component activation.
+     */
+    public void activate() {
+        super.activate(new HashMap<>());
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServiceCallback) {
+        this.discoveryServiceCallback = discoveryServiceCallback;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting SomfyTahoma background discovery");
+
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(this::runDiscovery, 10, DISCOVERY_REFRESH_SEC,
+                    TimeUnit.MINUTES);
+        }
+
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping SomfyTahoma background discovery");
+        if (discoveryJob != null && !discoveryJob.isCancelled()) {
+            discoveryJob.cancel(true);
+            discoveryJob = null;
+        }
+    }
+
+    @Override
+    public boolean isBackgroundDiscoveryEnabled() {
```

Comment:
Why do you override this? The super implementation allows background discovery to be deactivated as well.

Reply:
Removed overided method